InternshipProject#2/Manager/WatcherManager.cs

```diff
@@ -9,32 +9,46 @@ namespace InternshipProject_2.Manager;
 public class WatcherManager : IWatcherManager
 {
     private Project2Context _dbContext;
-    public HttpContext httpContext;
-
+    private readonly Mapper map;
     public WatcherManager(Project2Context dbContext)
     {
         _dbContext = dbContext;
+        map = MapperConfig.InitializeAutomapper();
     }
-    public async Task<WatchResponse> WatchTicket(WatchRequest request, int userId)
+    public async Task<WatchResponse> WatchTicket(WatchRequest request, int? userId)
     {
-        var existingWatchRequest = await _dbContext.Watchers
-        .FirstOrDefaultAsync(w => w.UserId == userId && w.TicketId == request.TicketId);
+        var isDeleted = await _dbContext.Watchers
```

Comment:
You don't need to make a new search to determine whether it is deleted or not, simply check its property: dbWatchTicket.IsDeleted.
The dbWatchTicket is a very odd name... watcher or ticketWatcher sounds much better.

Reply:
Deleted the useless search and renamed